src/controllers/DrawingModeController.js

```diff
@@ -157,12 +159,11 @@ class DrawingModeController extends AnnotationModeController {
             case 'softcommit':
                 this.currentThread = undefined;
                 this.saveThread(thread);
+                this.unbindListeners();
+                this.bindListeners();
```

Comment:
Why do we need to bind and unbind?

Reply:
Drawing handlers are bound to a temporary thread so this resets the handlers and essentially begins a new temp thread.